src/dashboard/Data/Browser/AttachSelectedRowsDialog.react.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import FormModal from 'components/FormModal/FormModal.react';
+import Field from 'components/Field/Field.react';
+import Label from 'components/Label/Label.react';
+import TextInput from 'components/TextInput/TextInput.react';
+import Dropdown from 'components/Dropdown/Dropdown.react';
+import Option from 'components/Dropdown/Option.react';
+import Parse from 'parse';
+import { SpecialClasses } from 'lib/Constants';
+
+export default class AttachSelectedRowsDialog extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentClass: null,
+      currentColumn: null,
+      touchableColumns: [],
+      targetObjectId: '',
+      objectIds: [],
+    };
+
+    this.handleConfirm = this.handleConfirm.bind(this);
+    this.handleClassChange = this.handleClassChange.bind(this);
+    this.handleColumnChange = this.handleColumnChange.bind(this);
+    this.handleTargetObjectIdChange = this.handleTargetObjectIdChange.bind(this);
+  }
+
+  componentWillMount() {
+    const { selection, classes, onSelectClass } = this.props;
+    if (selection) {
+      const currentClass = classes[0];
+      const objectIds = [];
+      for (const objectId in selection) {
+        objectIds.push(objectId);
+      }
+      const touchableColumns = onSelectClass(currentClass);
+      const currentColumn = touchableColumns[0];
+      this.setState({
+        currentClass: currentClass,
+        touchableColumns,
+        currentColumn,
+        objectIds,
+      });
+    }
+  }
+
+  handleTargetObjectIdChange(targetObjectId) {
+    this.setState({ targetObjectId });
+  }
+
+  handleConfirm() {
+    const {
+      currentClass,
+      currentColumn,
+      targetObjectId,
+      objectIds,
+    } = this.state;
+    const promise = new Parse.Promise();
+    this.props.onConfirm(currentClass, targetObjectId, currentColumn, objectIds)
+    .then(() => promise.resolve())
+    .catch((error) => promise.reject(error));
+    return promise;
```

Comment:
Same here, this can just be `this.props.onConfirm(currentClass, targetObjectId, currentColumn, objectIds)`